JuiceMaker/JuiceMaker/Model/JuiceMaker.swift

```diff
@@ -4,51 +4,82 @@
 //  Copyright © yagom academy. All rights reserved.
 //
 
+protocol JuiceMakerDelegate {
+    func successJuiceMake(_ menu: JuiceMaker.Menu)
+    func failJuiceMake()
+    func changeFruitStock(fruit: Fruit, amount: String)
+}
+
 struct JuiceMaker {
-    enum Menu {
+    enum Menu: Int {
         case strawberryJuice
         case bananaJuice
         case pineappleJuice
         case kiwiJuice
         case mangoJuice
         case strawberryAndBananaJuice
         case mangoAndKiwiJuice
-    }
-    
-    typealias Recipe = [(fruit: Fruit, amount: Int)]
-    private let store: FruitStore
-    
-    init(fruitStore: FruitStore) {
-        self.store = fruitStore
-    }
-
-    private func provideRecipe(_ menu: Menu) -> Recipe {
-        switch menu {
+        
+        var koreanName: String {
+            switch self {
             case .strawberryJuice:
-                return [(.strawberry, 16)]
+                return "딸기쥬스"
             case .bananaJuice:
-                return [(.banana, 2)]
+                return "바나나쥬스"
             case .pineappleJuice:
-                return [(.pineapple, 2)]
+                return "파인애플쥬스"
             case .kiwiJuice:
-                return [(.kiwi, 3)]
+                return "키위쥬스"
             case .mangoJuice:
-                return [(.mango, 3)]
+                return "망고쥬스"
             case .strawberryAndBananaJuice:
-                return [(.strawberry, 10), (.banana, 1)]
+                return "딸바쥬스"
             case .mangoAndKiwiJuice:
-                return [(.mango, 2), (.kiwi, 1)]
+                return "망키쥬스"
+            }
         }
     }
     
-    func canMakeJuice(menu: Menu) -> Bool {
-        let recipe = provideRecipe(menu)
+    typealias Recipe = [(fruit: Fruit, amount: Int)]
+    private let store: FruitStore
+    private let recipe: [Menu: Recipe]
+    var delegate: JuiceMakerDelegate?
+    
+    init(_ fruitStore: FruitStore, _ recipe: [Menu: Recipe]) {
+        self.store = fruitStore
+        self.recipe = recipe
+    }
+
+    private func provideRecipe(_ menu: Menu) -> Recipe? {
+        guard let juiceRecipe = recipe[menu] else { return nil }
+        
+        return juiceRecipe
+    }
+    
+    func makeJuice(menu: Menu) {
+        guard let recipe = provideRecipe(menu) else { return }
+        
+        guard canMakeJuice(recipe) else {
+            delegate?.failJuiceMake()
+            return
+        }
         
+        consumeFruit(recipe)
+        delegate?.successJuiceMake(menu)
+    }
+    
+    private func canMakeJuice(_ recipe: Recipe ) -> Bool {
         guard recipe.allSatisfy({ fruit, amount in return store.isEnoughFruits(fruit, count: amount) }) else { return false }
-
+        
+        return true
```

Comment:
```suggestion
        return recipe.allSatisfy({ fruit, amount in return store.isEnoughFruits(fruit, count: amount) })
```

지난 번과는 로직이 조금 달라졌는데, 그렇다면 이렇게 바로 `return`에 넣어 반환하는 것은 어떨까요?

Reply:
수정하였습니다 😎